test/simulator/timestamp/simulator_interface/simulator_interface.cpp

```diff
@@ -0,0 +1,185 @@
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "simulator_interface.h"
+
+#include <iostream>
+#include <sstream>
+#include <string>
+
+void
+print_border_msg(const std::string &msg, const std::string &color)
+{
+    const int count = msg.length() + 2;
+    const std::string dash(count, '-');
+
+    std::cout << color << "+" + dash + "+" << RESET << std::endl;
+    std::cout << color << "| " << msg << " |" << RESET << std::endl;
+    std::cout << color << "+" + dash + "+" << RESET << std::endl;
+}
+
+int
+choose_num(int min, int max, const std::string &cli_str)
+{
+    std::string text_line;
+    int choice;
+
+    do {
+        std::cout << "\n" << cli_str << " ";
+        std::getline(std::cin, text_line);
+
+        /* Extract number from the text. */
+        std::istringstream text_stream(text_line);
+        text_stream >> choice;
+
+        /* Validate the number of choice. */
+        if (choice < min || choice > max)
+            print_border_msg(
+              "Choose a number between " + std::to_string(min) + " and " + std::to_string(max),
+              RED);
+
+    } while (choice < min || choice > max);
+
+    return (choice);
+}
+
+void
+interface_session_management()
+{
+}
+
+void
+interface_set_timestamp()
+{
+}
+
+void
+interface_conn_query_timestamp()
+{
+}
+
+void
+interface_begin_transaction()
+{
+}
+
+void
+interface_commit_transaction()
+{
+}
+
+void
+interface_prepare_transaction()
+{
+}
+
+void
+interface_rollback_transaction()
+{
+}
+
+void
+interface_timestamp_transaction()
+{
+}
+
+void
+interface_session_query_timestamp()
+{
+}
+
+void
+print_rules()
+{
+}
+
+int
+main(int argc, char *argv[])
+{
+    bool exit = false;
+
+    do {
+        std::cout << std::endl;
+
+        std::cout << "1: Session Management " << std::endl;
+        std::cout << "2: [Conn] set_timestamp()" << std::endl;
+        std::cout << "3: [Conn] query_timestamp()" << std::endl;
+        std::cout << "4: [Session] begin_transaction()" << std::endl;
+        std::cout << "5: [Session] commit_transaction()" << std::endl;
+        std::cout << "6: [Session] prepare_transaction()" << std::endl;
+        std::cout << "7: [Session] rollback_transaction()" << std::endl;
+        std::cout << "8: [Session] timestamp_transaction()" << std::endl;
+        std::cout << "9: [Session] query_timestamp()" << std::endl;
+        std::cout << "10: print rules for timestamps" << std::endl;
+        std::cout << "11: exit" << std::endl;
```

Comment:
I think a reset or restart option would be useful.

Reply:
Good idea, I will probably get to this at the end of the PR.

Reply:
I think a reset option can be added later if this tool is used often. Rn, it does not seem worth the effort. 
If someone wants to reset, they can exit and start over.

Reply:
Yeah that's fine